build/utils.js

```diff
@@ -30,42 +29,38 @@ function getControllerDir(isInstall) {
     // As a fallback solution, we walk up the directory tree until we reach the root or find js-controller
     // Apparently, checking vs null/undefined may miss the odd case of controllerPath being ""
     // Thus we check for falsyness, which includes failing on an empty path
-    if (!controllerPath) {
-        let curDir = path.join(__dirname, "../node_modules");
-        let parentDir;
-        outer: while (true) {
-            for (const pkg of possibilities) {
-                const possiblePath = path.join(curDir, pkg);
-                try {
-                    if (fs.existsSync(possiblePath)) {
-                        controllerPath = possiblePath;
-                        break outer;
-                    }
-                }
-                catch (_b) {
-                    // don't care
+    // We start in the node_modules subfolder of adapter-core, which is the deepest we should be able to expect the controller
+    let curDir = path.join(__dirname, "../node_modules");
+    while (true) {
+        for (const pkg of possibilities) {
+            const possiblePath = path.join(curDir, pkg, "package.json");
+            try {
+                // If package.json exists in the directory and its name field matches, we've found js-controller
+                if (fs.existsSync(possiblePath) &&
+                    JSON.parse(fs.readFileSync(possiblePath, "utf8")).name ===
+                        pkg.toLowerCase()) {
+                    return path.dirname(possiblePath);
```

Comment:
@foxriver76 I thought it makes more sense to look for package.json here and check its contents instead of hoping that iob.bat isn't the wrong directory.

Reply:
I don't get the linkage to `iob.bat` right now.

Reply:
I also really don't hope that this check is necessary for any installation, but yes never know what users do.

Reply:
For reference, this is what I meant: https://github.com/ioBroker/ioBroker.js-controller/blob/bf8adb77142103163b7aacca22a8555303aa9e1b/packages/common/src/lib/common/tools.ts#L2119

Reply:
ah now I understand. Yes, package.json is guranteed to be in root package, I also think it is more save than `iob.bat`